<?xml version="1.0" encoding="UTF-8"?>
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
-->
<robot name="right_srm">
    <group name="srm_base_r">
        <joint name="right_base_joint"/>
        <joint name="right_summit_back_left_wheel_joint"/>
        <joint name="right_summit_back_right_wheel_joint"/>
        <joint name="right_summit_front_left_wheel_joint"/>
        <joint name="right_summit_front_right_wheel_joint"/>
        <joint name="right_summit_base_footprint_joint"/>
    </group>
    <group name="srm_r_ur">
        <joint name="ur_right_base_link-base_link_inertia"/>
        <joint name="ur_right_shoulder_pan_joint"/>
        <joint name="ur_right_shoulder_lift_joint"/>
        <joint name="ur_right_elbow_joint"/>
        <joint name="ur_right_wrist_1_joint"/>
        <joint name="ur_right_wrist_2_joint"/>
        <joint name="ur_right_wrist_3_joint"/>
        <chain base_link="ur_right_base_link_inertia" tip_link="ur_right_tool0"/>
    </group>

    <group name="srm_r_ee">
        <link name="ur_right_paletta"/>
    </group>

    <group name="srm_r">
        <group name="srm_base_r"/>
        <group name="srm_r_ur"/>
        <group name="srm_r_ee"/>
    </group>

    <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
    <group_state name="home_r" group="srm_r_ur">
<!-- 
        <joint name="right_summit_back_left_wheel_joint" value="0"/>
        <joint name="right_summit_back_right_wheel_joint" value="0"/>
        <joint name="right_summit_front_left_wheel_joint" value="0"/>
        <joint name="right_summit_front_right_wheel_joint" value="0"/>
         -->
        <joint name="ur_right_elbow_joint" value="-2.6609"/>
        <joint name="ur_right_shoulder_lift_joint" value="0.268"/>
        <joint name="ur_right_shoulder_pan_joint" value="-1.57"/>
        <joint name="ur_right_wrist_1_joint" value="-1.3893"/>
        <joint name="ur_right_wrist_2_joint" value="1.3893"/>
        <joint name="ur_right_wrist_3_joint" value="-0.1694"/>
    </group_state>
    
    <!-- Group state of the complete block base + ur + ee-->
    <group_state name="home_r_base_ur_ee" group="srm_r">
        <!-- <joint name="right_summit_back_left_wheel_joint" value="0"/>
        <joint name="right_summit_back_right_wheel_joint" value="0"/>
        <joint name="right_summit_front_left_wheel_joint" value="0"/>
        <joint name="right_summit_front_right_wheel_joint" value="0"/>
     -->
        <joint name="ur_right_elbow_joint" value="-2.6609"/>
        <joint name="ur_right_shoulder_lift_joint" value="0.268"/>
        <joint name="ur_right_shoulder_pan_joint" value="-1.57"/>
        <joint name="ur_right_wrist_1_joint" value="-1.3893"/>
        <joint name="ur_right_wrist_2_joint" value="1.3893"/>
        <joint name="ur_right_wrist_3_joint" value="-0.1694"/>
        <!-- <joint name="ur_right_base_link-base_link_inertia" value="0"/> -->
    </group_state>
    
    <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
    <end_effector name="ee_r" parent_link="ur_right_tool0" group="srm_r_ee" parent_group="srm_r_ur"/>
    
    <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->

    <disable_collisions link1="right_summit_back_left_wheel_link" link2="right_summit_back_right_wheel_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_left_wheel_link" link2="right_summit_base_link" reason="Adjacent"/>
    <disable_collisions link1="right_summit_back_left_wheel_link" link2="right_summit_front_laser_base_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_left_wheel_link" link2="right_summit_front_left_wheel_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_left_wheel_link" link2="right_summit_front_right_wheel_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_left_wheel_link" link2="right_summit_rear_laser_base_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_left_wheel_link" link2="ur_right_base_link_inertia" reason="Never"/>
    <disable_collisions link1="right_summit_back_left_wheel_link" link2="ur_right_shoulder_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_left_wheel_link" link2="ur_right_upper_arm_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_right_wheel_link" link2="right_summit_base_link" reason="Adjacent"/>
    <disable_collisions link1="right_summit_back_right_wheel_link" link2="right_summit_front_laser_base_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_right_wheel_link" link2="right_summit_front_left_wheel_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_right_wheel_link" link2="right_summit_front_right_wheel_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_right_wheel_link" link2="right_summit_rear_laser_base_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_right_wheel_link" link2="ur_right_base_link_inertia" reason="Never"/>
    <disable_collisions link1="right_summit_back_right_wheel_link" link2="ur_right_shoulder_link" reason="Never"/>
    <disable_collisions link1="right_summit_back_right_wheel_link" link2="ur_right_upper_arm_link" reason="Never"/>
    <disable_collisions link1="right_summit_base_link" link2="right_summit_front_laser_base_link" reason="Adjacent"/>
    <disable_collisions link1="right_summit_base_link" link2="right_summit_front_left_wheel_link" reason="Adjacent"/>
    <disable_collisions link1="right_summit_base_link" link2="right_summit_front_right_wheel_link" reason="Adjacent"/>
    <disable_collisions link1="right_summit_base_link" link2="right_summit_rear_laser_base_link" reason="Adjacent"/>
    <disable_collisions link1="right_summit_base_link" link2="ur_right_base_link_inertia" reason="Adjacent"/>
    <disable_collisions link1="right_summit_base_link" link2="ur_right_shoulder_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_laser_base_link" link2="right_summit_front_left_wheel_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_laser_base_link" link2="right_summit_front_right_wheel_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_laser_base_link" link2="right_summit_rear_laser_base_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_laser_base_link" link2="ur_right_base_link_inertia" reason="Never"/>
    <disable_collisions link1="right_summit_front_laser_base_link" link2="ur_right_shoulder_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_left_wheel_link" link2="right_summit_front_right_wheel_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_left_wheel_link" link2="right_summit_rear_laser_base_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_left_wheel_link" link2="ur_right_base_link_inertia" reason="Never"/>
    <disable_collisions link1="right_summit_front_left_wheel_link" link2="ur_right_shoulder_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_left_wheel_link" link2="ur_right_upper_arm_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_right_wheel_link" link2="right_summit_rear_laser_base_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_right_wheel_link" link2="ur_right_base_link_inertia" reason="Never"/>
    <disable_collisions link1="right_summit_front_right_wheel_link" link2="ur_right_shoulder_link" reason="Never"/>
    <disable_collisions link1="right_summit_front_right_wheel_link" link2="ur_right_upper_arm_link" reason="Never"/>
    <disable_collisions link1="right_summit_rear_laser_base_link" link2="ur_right_base_link_inertia" reason="Never"/>
    <disable_collisions link1="right_summit_rear_laser_base_link" link2="ur_right_shoulder_link" reason="Never"/>
    <disable_collisions link1="ur_right_base_link_inertia" link2="ur_right_shoulder_link" reason="Adjacent"/>
    <disable_collisions link1="ur_right_forearm_link" link2="ur_right_upper_arm_link" reason="Adjacent"/>
    <disable_collisions link1="ur_right_forearm_link" link2="ur_right_wrist_1_link" reason="Adjacent"/>
    <disable_collisions link1="ur_right_forearm_link" link2="ur_right_wrist_2_link" reason="Never"/>
    <disable_collisions link1="ur_right_paletta" link2="ur_right_wrist_1_link" reason="Never"/>
    <disable_collisions link1="ur_right_paletta" link2="ur_right_wrist_2_link" reason="Never"/>
    <disable_collisions link1="ur_right_paletta" link2="ur_right_wrist_3_link" reason="Adjacent"/>
    <disable_collisions link1="ur_right_shoulder_link" link2="ur_right_upper_arm_link" reason="Adjacent"/>
    <disable_collisions link1="ur_right_wrist_1_link" link2="ur_right_wrist_2_link" reason="Adjacent"/>
    <disable_collisions link1="ur_right_wrist_1_link" link2="ur_right_wrist_3_link" reason="Never"/>
    <disable_collisions link1="ur_right_wrist_2_link" link2="ur_right_wrist_3_link" reason="Adjacent"/>
</robot>